src/public-modules/Bounties/sagas.js

```diff
@@ -28,13 +31,82 @@ const {
   loadBountiesFail,
   loadBountiesSuccess,
   loadMoreBountiesFail,
-  loadMoreBountiesSuccess
+  loadMoreBountiesSuccess,
+  setSearch,
+  toggleStageFilter,
+  setStageFilter,
+  toggleDifficultyFilter,
+  setDifficultyFilter,
+  addCategoryFilter,
+  removeCategoryFilter,
+  resetFilter
 } = actions;
 
+export function* initializeFiltersFromQuery() {
+  const params = queryStringToObject(window.location.search);
+
+  const { search, bountyStage, difficulty, category } = params;
+
+  if (search) {
+    yield put(resetFilter('search'));
+    yield put(setSearch(search));
+  }
+
+  if (difficulty === '') {
+    yield put(resetFilter('difficulty'));
+  }
+
+  if (difficulty) {
+    yield put(resetFilter('difficulty'));
+    const difficulties = difficulty.split(',');
+    for (let i = 0; i < difficulties.length; i++) {
+      yield put(toggleDifficultyFilter(difficulties[i]));
+    }
+  }
+
+  if (bountyStage === '') {
+    yield put(resetFilter('stage'));
+  }
+
+  if (bountyStage) {
+    yield put(resetFilter('stage'));
+    const stages = bountyStage.split(',');
+    for (let i = 0; i < stages.length; i++) {
+      yield put(toggleStageFilter(stages[i]));
+    }
+  }
+
+  if (category === '') {
+    yield put(resetFilter('category'));
+  }
+
+  if (category) {
+    yield put(resetFilter('category'));
+    const categories = category.split(',');
+    for (let i = 0; i < categories.length; i++) {
+      yield put(addCategoryFilter(categories[i]));
+    }
+  }
```

Comment:
@c-o-l-o-r we need to add this for sort and also for the issuer filters. I'll let you do that so you can dive in a little deeper :)